cea/analysis/uncertainty/Uncertainty_parameters.py

```diff
@@ -0,0 +1,266 @@
+" This file is used to generate various scenarios which involve uncertainty. Go through README.md"
+
+__author__ = "Sreepathi Bhargava Krishna"
+__copyright__ = "Copyright 2015, Architecture and Building Systems - ETH Zurich"
+__credits__ = ["Sreepathi Bhargava Krishna"]
+__license__ = "MIT"
+__version__ = "0.1"
+__maintainer__ = "Daren Thomas"
+__email__ = "cea@arch.ethz.ch"
+__status__ = "Production"
+
+def run_as_script(scenario_path):
+    import pandas as pd
+    import cea.inputlocator
+    import numpy as np
+
+    # Importing the excel sheet containing the uncertainty distributions
+    locator = cea.inputlocator.InputLocator(scenario_path)
+    uncertainty_db = pd.read_excel(locator.get_uncertainty_db(), 'ECONOMIC')
+
+    # Fetching the distribution type and the corresponding variables from the excel sheet
+    HP_n_distribution = uncertainty_db[uncertainty_db['name'] == 'HP_n']['distribution'].max()  # lifetime [years] distribution 20
+    GHP_nHP_distribution = uncertainty_db[uncertainty_db['name'] == 'GHP_nHP']['distribution'].max()  # for the geothermal heat pump default 20
+    Boiler_n_distribution = uncertainty_db[uncertainty_db['name'] == 'Boiler_n']['distribution'].max()  # lifetime, after A+W, default 20
+    CC_n_distribution = uncertainty_db[uncertainty_db['name'] == 'CC_n']['distribution'].max()  # lifetime default 25
+    FC_n_distribution = uncertainty_db[uncertainty_db['name'] == 'FC_n']['distribution'].max()  # years of operation default 10
+    PVT_n_distribution = uncertainty_db[uncertainty_db['name'] == 'PVT_n']['distribution'].max()  # years of operation default 20
+    CT_a_distribution = uncertainty_db[uncertainty_db['name'] == 'CT_a']['distribution'].max()  # annuity factor default 0.15
+    Subst_n_distribution = uncertainty_db[uncertainty_db['name'] == 'Subst_n']['distribution'].max()  # Lifetime after A+W default 20
+    ELEC_PRICE_distribution = uncertainty_db[uncertainty_db['name'] == 'ELEC_PRICE']['distribution'].max()  # default 0.2
+    NG_PRICE_distribution = uncertainty_db[uncertainty_db['name'] == 'NG_PRICE']['distribution'].max()  # [CHF / wh] # default 0.068
+    BG_PRICE_distribution = uncertainty_db[uncertainty_db['name'] == 'BG_PRICE']['distribution'].max()   # [CHF / wh] # default 0.076
+    Subst_i_distribution = uncertainty_db[uncertainty_db['name'] == 'Subst_i']['distribution'].max()  # default 0.05
+    FC_i_distribution = uncertainty_db[uncertainty_db['name'] == 'FC_i']['distribution'].max()  # interest rate default 0.05
+    HP_i_distribution = uncertainty_db[uncertainty_db['name'] == 'HP_i']['distribution'].max()  # interest rate default 0.05
+    Boiler_i_distribution = uncertainty_db[uncertainty_db['name'] == 'Boiler_i']['distribution'].max()  # interest rate default 0.05
+
+    if HP_n_distribution == 'Beta':
```

Comment:
This way of coding encapsulates the variables in the code. So at the moment I add a new variable to the database the code will not be able to read it. better, create a dictionary automatically.

Check list comprehensions in python as they can be handy in this. check the lines of code 70 -75 of cea\analysis\sensitivity\sensitivity_demand_samples.py for an example.

It is worth doing it as it can reduce lines 2 - 230 into just 10 lines of code!